authorization/role/repository/default_role_mapping_blackbox_test.go

```diff
@@ -0,0 +1,148 @@
+package repository_test
+
+import (
+	"testing"
+
+	rolerepo "github.com/fabric8-services/fabric8-auth/authorization/role/repository"
+	"github.com/fabric8-services/fabric8-auth/gormtestsupport"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type defaultRoleMappingBlackBoxTest struct {
+	gormtestsupport.DBTestSuite
+	repo rolerepo.DefaultRoleMappingRepository
+}
+
+func TestRunDefaultRoleMappingBlackBoxTest(t *testing.T) {
+	suite.Run(t, &defaultRoleMappingBlackBoxTest{DBTestSuite: gormtestsupport.NewDBTestSuite()})
+}
+
+func (s *defaultRoleMappingBlackBoxTest) SetupTest() {
+	s.DBTestSuite.SetupTest()
+	s.repo = rolerepo.NewDefaultRoleMappingRepository(s.DB)
+}
+
+func (s *defaultRoleMappingBlackBoxTest) TestOKToDelete() {
+	g := s.NewTestGraph()
+
+	rt := g.CreateResourceType()
+	rm := rolerepo.DefaultRoleMapping{
+		ResourceTypeID: rt.ResourceType().ResourceTypeID,
+		FromRoleID:     g.CreateRole(rt).Role().RoleID,
+		ToRoleID:       g.CreateRole().Role().RoleID,
+	}
+
+	err := s.repo.Create(s.Ctx, &rm)
+	require.NoError(s.T(), err)
+
+	rt2 := g.CreateResourceType()
+	rm2 := rolerepo.DefaultRoleMapping{
+		ResourceTypeID: rt2.ResourceType().ResourceTypeID,
+		FromRoleID:     g.CreateRole(rt).Role().RoleID,
+		ToRoleID:       g.CreateRole().Role().RoleID,
+	}
+
+	err = s.repo.Create(s.Ctx, &rm2)
+	require.NoError(s.T(), err)
+
+	mappings, err := s.repo.List(s.Ctx)
+	require.Nil(s.T(), err, "Could not list role mappings")
+
+	found1 := false
+	found2 := false
+
+	for _, mapping := range mappings {
+		if mapping.DefaultRoleMappingID == rm.DefaultRoleMappingID {
+			found1 = true
+		} else if mapping.DefaultRoleMappingID == rm2.DefaultRoleMappingID {
+			found2 = true
+		}
+	}
+
+	require.True(s.T(), found1, "first default role mapping not found")
+	require.True(s.T(), found2, "second default role mapping not found")
+
+	err = s.repo.Delete(s.Ctx, rm.DefaultRoleMappingID)
+	assert.Nil(s.T(), err)
+
+	mappings, err = s.repo.List(s.Ctx)
+	require.Nil(s.T(), err, "Could not list role mappings")
+
+	for _, data := range mappings {
+		// The default role mapping rm was deleted while rm2 was not deleted, hence we check
+		// that none of the role mappings returned include the deleted record.
+		require.NotEqual(s.T(), rm.DefaultRoleMappingID.String(), data.DefaultRoleMappingID.String())
+	}
+}
+
+func (s *defaultRoleMappingBlackBoxTest) TestOKToLoad() {
+	g := s.NewTestGraph()
+	rt := g.CreateResourceType()
+	rm := &rolerepo.DefaultRoleMapping{
+		ResourceTypeID: rt.ResourceType().ResourceTypeID,
+		FromRoleID:     g.CreateRole(rt).Role().RoleID,
+		ToRoleID:       g.CreateRole().Role().RoleID,
+	}
+
+	err := s.repo.Create(s.Ctx, rm)
+	require.NoError(s.T(), err)
+
+	_, err = s.repo.Load(s.Ctx, rm.DefaultRoleMappingID)
+	require.NoError(s.T(), err)
```

Comment:
Check you loaded the expected role mapping.

Reply:
Also check that load() for unknown ID returns NotFoundError.

Reply:
Fixed

Reply:
I've created a dedicated test utility function for such tests recently.
So, you can use something like:
`testsupport.AssertError(s.T(), err, errors.NotFoundError{}, "default_role_mapping with id '%s' not found", id.String())`
It will check if the error in nil or not. Also will check the error type and the message.